app/src/main/kotlin/games/Board.kt

```diff
@@ -41,22 +41,42 @@ class Board :ValidMoveProvider{
         elements[move.row][move.column].setState(player.id)
     }
 
+    internal fun setState(gameState: String) {
+        val it = gameState.splitToSequence(" ", "\n").iterator()
+        for (r in 0..2) {
+            for (c in 0..2) {
+                elements[r][c].setState(toState(it.next()))
+            }
+        }
+    }
+
     fun empty(move: Move) {
         elements[move.row][move.column].setState(0)
     }
 
-    // add a fun undo which reverts to the former state.
-
-    //    val elements = arrayOf(arrayOf(BoardElement(), BoardElement(), BoardElement()))
-    val size = 3
-    val row1 = List(size){
-        BoardElement()
-    }
-    val row2 = List(size){
-        BoardElement()
-    }
-    val row3 = List(size){
-        BoardElement()
+    private fun toState(symbol: String): Int {
+        return when (symbol) {
+            "." -> 0
+            "X" -> 1
+            "O" -> 2
+            else -> throw Exception("unrecognized symbol $symbol")
+        }
     }
+
+    val row1 = listOf(
+            BoardElement(0, 0),
+            BoardElement(0, 1),
+            BoardElement(0, 2)
+    )
+    val row2 = listOf(
+            BoardElement(1, 0),
+            BoardElement(1, 1),
+            BoardElement(1, 2)
+    )
+    val row3 = listOf(
+            BoardElement(2, 0),
+            BoardElement(2, 1),
+            BoardElement(2, 2)
+    )
```

Comment:
Moving the location on the board into the board elements caused a rewrite of the board initialization code.  There's probably a better way to do this.